stripe.go

```diff
@@ -520,51 +520,56 @@ func (s *BackendImplementation) ResponseToError(res *http.Response, resBody []by
 	if s.Type == ConnectBackend {
 		// If this is an OAuth request, deserialize as Error because OAuth errors
 		// are a different shape from the standard API errors.
-		var topLevelError rawErrorInternal
+		var topLevelError Error
 		if err := s.UnmarshalJSONVerbose(res.StatusCode, resBody, &topLevelError); err != nil {
 			return err
 		}
-		raw.E = &topLevelError
+		raw.Error = &topLevelError
 	} else {
 		if err := s.UnmarshalJSONVerbose(res.StatusCode, resBody, &raw); err != nil {
 			return err
 		}
 	}
 
 	// no error in resBody
-	if raw.E == nil {
+	if raw.Error == nil {
 		err := errors.New(string(resBody))
 		return err
 	}
-	raw.E.HTTPStatusCode = res.StatusCode
-	raw.E.RequestID = res.Header.Get("Request-Id")
+	raw.Error.HTTPStatusCode = res.StatusCode
+	raw.Error.RequestID = res.Header.Get("Request-Id")
 
 	var typedError error
-	switch raw.E.Type {
+	switch raw.Error.Type {
 	case ErrorTypeAPI:
-		typedError = &APIError{stripeErr: raw.E.Error}
+		typedError = &APIError{stripeErr: raw.Error}
```

Comment:
All the changes in here are basically because I "flattened" the error types by killing `rawErrorInternal`. I also renamed `rawError.E` to `rawError.Error` to make it a little more legible.